src/rpm/package.rs

```diff
@@ -426,52 +582,113 @@ impl RPMPackageMetadata {
                             .map(|file_size| file_size as _)
                             .collect::<Vec<u64>>()
                     })
-            })?;
+            });
         let flags = self
             .header
-            .get_entry_data_as_u32_array(IndexTag::RPMTAG_FILEFLAGS)?;
+            .get_entry_data_as_u32_array(IndexTag::RPMTAG_FILEFLAGS);
         // @todo
         // let caps = self.get_entry_i32_array_data(IndexTag::RPMTAG_FILECAPS)?;
 
-        let paths = self.get_file_paths()?;
-        let n = paths.len();
-
-        let v = itertools::multizip((
-            paths.into_iter(),
-            users,
-            groups,
-            modes,
-            digests,
-            mtimes,
-            sizes,
-            flags,
-        ))
-        .try_fold::<Vec<FileEntry>, _, Result<_, RPMError>>(
-            Vec::with_capacity(n),
-            |mut acc, (path, user, group, mode, digest, mtime, size, flags)| {
-                let digest = if digest.is_empty() {
-                    None
-                } else {
-                    Some(FileDigest::load_from_str(algorithm, digest)?)
-                };
-                let utc = chrono::Utc;
-                acc.push(FileEntry {
-                    path,
-                    ownership: FileOwnership {
-                        user: user.to_owned(),
-                        group: group.to_owned(),
+        match (modes, users, groups, digests, mtimes, sizes, flags) {
+            (Ok(modes), Ok(users), Ok(groups), Ok(digests), Ok(mtimes), Ok(sizes), Ok(flags)) => {
+                let paths = self.get_file_paths()?;
+                let n = paths.len();
+
+                let v = itertools::multizip((
```

Comment:
Is it worth adding comments to this bit of code? Just to explain what it's trying to do.

Reply:
I think it's fairly clear, but I wouldn't complain about more comments being added